### PR TITLE
VSCode hardware debugging

### PR DESCRIPTION
**Pull request by hamishwillee**
Adds section on hardware debugging

### CODE REVIEW

File: en/setup/vscode.md [hunk 1]
```diff
@@ -73,6 +75,12 @@ To debug PX4 on SITL:
 
 While debugging you can set breakpoints, step over code, and otherwise develop as normal.
 
+### Hardware Debugging
+
+The instructions in [SWD (JTAG) Hardware Debugging Interface](../debug/swd_debug.html) explain how to connect to the SWD interface on common flight controllers (for example, using the Dronecode or Blackmagic probes).
+
+After connecting to the SWD interface, hardware debugging in VSCode is then typically the same as for [SITL Debugging](#debugging_sitl) except you select the debug target for your firmware - e.g. `px4_fmu-v5`.
```
Comment (hamishwillee) [inline en/setup/vscode.md]: @julianoes @dagar This is a complete guess - is there more to setting up VSCode for hardware debugging than this?

Comment (hamishwillee) [reply en/setup/vscode.md]: @julianoes ?

Comment (julianoes) [reply en/setup/vscode.md]: Sorry no idea. I'm ignoring this for now.

Comment (dagar) [reply en/setup/vscode.md]: That's mostly correct, but there are different options per debugger type.

Screen Shot 2020-10-27 at 7 54 56 PM

**Comment by hamishwillee** [comment]
Thanks. Merged with additional image/info from dagar. Might not be 100% correct (not tested by me) but better than what was present.

**Pull request merged**